Storage/SQL/SQLiteBookmarksSyncing.swift

```diff
@@ -438,8 +438,11 @@ public class SQLiteBookmarkBufferStorage: BookmarkBufferStorage {
             "folderName = ?, queryId = ? " +
             "WHERE guid = ?"
 
+            // We used to use INSERT OR IGNORE here, but it muffles legitimate errors. The only
+            // real use for that is/was to catch duplicates, but the UPDATE we run first should
+            // serve that purpose just as well.
             let insert =
-            "INSERT OR IGNORE INTO \(TableBookmarksBuffer) " +
```

Comment:
Yes. The general idea is: the `UPDATE` above will catch the common usage of `INSERT OR IGNORE` — "insert this unless it already exists". And after that, there's basically no case in which we will be best served by dropping an incoming record on the floor — we have no mechanism to go back and re-fetch such a malformed record.

The path now will be a failure to buffer, followed by a repeated redownload until success. We're betting that won't happen very often, because we're also in parallel loosening the constraints on incoming records (and I'll push a commit shortly that allows ordinary records to be missing a parent title).